jxl/src/render/stages/patches.rs

```diff
@@ -58,6 +58,65 @@ impl RenderPipelineStage for PatchesStage {
 
 #[cfg(test)]
 mod test {
+    use test_log::test;
+
+    use super::*;
+    use crate::error::Result;
+    use crate::features::patches::{
+        PatchBlendMode, PatchBlending, PatchPosition, PatchReferencePosition, PatchTreeNode,
+    };
+    use crate::util::test::read_headers_and_toc;
+
     #[test]
-    fn process_row_chunk() {}
+    fn patches_consistency() -> Result<()> {
+        let (file_header, _, _) =
+            read_headers_and_toc(include_bytes!("../../../resources/test/basic.jxl")).unwrap();
+        let patch_dict = PatchesDictionary {
```

Comment:
I would prefer if instead we had a cfg(test) PatchesDictionary::random constructor (and we didn't change the visibility), can we do that?

Reply:
How about this?